File: test/examples/pipe_operator.js
import Immutable from 'seamless-immutable';
// single line pipes not finished
const bar = _.chain(Immutable([1, 2, 3])).pipesCall(Enum.uppcase).value();

// [1, 2, 3] .pipesCall(Enum.uppcase).pipesCall(Enum.uppcase).value();
// [1, 2, 3] |> Enum.uppcase |> Enum.uppcase

const f = _.chain(Immutable([1, 2, 3])).pipesCall(Enum.uppcase(3, 2)).pipesCall(Enum.uppcase).value();
return _.chain(Immutable([5, 2, 3])).pipesCall(Enum.uppcase).value();
const b = _.chain(bar).pipesCall(Enum.uppcase).value();
return _.chain(bar).pipesCall(Enum.uppcase).value();


const foo = 20

return _.chain(foo)
.pipesCall(multi_no_parens, 10, 20)
.pipesCall(multi_has_parens, 10, 20)
.pipesCall(no_parens)
.pipesCall(empty_parens).value();


const res = _.chain(foo)
.pipesCall(has_parens, 10, 20)
.pipesCall(map, (x) => x * 2)
.pipesCall(map, ((x) => x * 2))
.pipesCall(map, (x => x * 2))
.pipesCall(map, x => x * 2).value();

File: test/examples/pipe_operator.rs
foo = 2

res = foo
|> add 10, 20
|> has_parens(10, 20)
|> noparens
|> emprty_parens()

res
|> add 10, 20
|> has_parens(10, 20)
|> map (x) => x * 2

|> emprty_parens()


return {foo: 1, bar: 2}
|> add 10, 20

[1, 2, 22]
|> add 10, 20

//|> square 2
//|> map (x) => x * 2
//|> square

puts res

puts "foo |> par"
puts 'foo |> par'

# |> square 2
# .thru(square, 2)

function square(x) {
  return x * x;
}
